crates/server/src/api/routes/quests/create_quest.rs

```diff
@@ -66,37 +103,19 @@ async fn create_quest_controller<DB: QuestsDatabase>(
     creator_address: &str
 ) -> Result<String, QuestError> {
     create_quest_req
-        .is_valid()
-        .map_err(|error| QuestError::QuestValidation(error.to_string()))?;
-
+        .is_valid()?; 
+    
     let quest = create_quest_req.to_create_quest()?;
     let id = db.create_quest(&quest, creator_address)
         .await
         .map_err(|_| QuestError::CommonError(CommonError::Unknown))?;
 
     if let Some(QuestReward { hook, items }) = &create_quest_req.reward {
-        if !is_url(&hook.webhook_url) {
-            return Err(QuestError::QuestValidation("Webhook url is not valid".to_string()));
-        }
-
-        if !items.is_empty() {
-            if !items.iter().all(|item| is_url(&item.image_link)) {
-                return Err(QuestError::QuestValidation("Item's image link is not valid".to_string()));
-            }
-    
-            if !items.iter().all(|item| item.name.len() > 3) {
-                return Err(QuestError::QuestValidation("Item name must be at least 3 characters".to_string()));
-            }
-
-            db.add_reward_hook_to_quest(&id, hook)
-            .await
-            .map_err(|_| QuestError::CommonError(CommonError::Unknown))?;
+        db.add_reward_hook_to_quest(&id, hook)
```

Comment:
I think we should use a transaction for the quest creation and the rewards

Reply:
yep, I'm agree